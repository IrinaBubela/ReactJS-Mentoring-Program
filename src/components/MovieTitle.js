import React from 'react';

const MovieTitle = ({ movie, onClick }) => {

  return (
    <div className="container d-flex align-items-center justify-content-center mt-4" onClick={() => onClick(movie)}>
      <div className="movie-details-container col text-center">
        <div className="poster">
          <img src={movie.imageUrl} alt={movie.name} />
        </div>
        <div className="details mt-4">
          <div className="name-and-genre">
            <h6>{movie.name}</h6>
            <p>{movie.genres.join(',')}</p>
          </div>
          <div className="year">
            <p className="border border-gray rounded p-1">{movie.releaseYear}</p>
          </div>
        </div>
      </div>
    </div>
  );
};

export default MovieTitle;
