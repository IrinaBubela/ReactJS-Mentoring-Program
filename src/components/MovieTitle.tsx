import React from 'react';
import { Movie } from './MovieListPage';

interface Props {
    movie: Movie;
    onClick: (movie: Movie) => void;
}

const MovieTitle: React.FC<Props> = ({ movie, onClick }) => {
  return (
    <div className="container d-flex align-items-center justify-content-center mt-4" onClick={() => onClick(movie)}>
      <div className="movie-details-container col text-center">
        <div className="poster">
          <img src={movie.poster_path} alt={movie.title} />
        </div>
        <div className="details mt-4">
          <div className="name-and-genre">
            <h6>{movie.title}</h6>
            <p>{movie.genres.join(',')}</p>
          </div>
          <div className="year">
            <p className="border border-gray rounded p-1">{movie.release_date}</p>
          </div>
        </div>
      </div>
    </div>
  );
};

export default MovieTitle;
