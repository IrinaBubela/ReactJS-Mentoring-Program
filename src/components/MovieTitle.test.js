import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import MovieTitle from './MovieTitle';

// Mock movie data
const movie = {
  imageUrl: 'example.jpg',
  name: 'Example Movie',
  genres: ['Action', 'Adventure'],
  releaseYear: 2021
};

describe('MovieTitle', () => {
  it('renders correctly', () => {
    const onClickMock = jest.fn();
    const { getByText, getByAltText } = render(<MovieTitle movie={movie} onClick={onClickMock} />);

    expect(getByText('Example Movie')).toBeInTheDocument();
    expect(getByText('Action,Adventure')).toBeInTheDocument();
    expect(getByText('2021')).toBeInTheDocument();

    const moviePoster = getByAltText('Example Movie');
    expect(moviePoster).toBeInTheDocument();
    expect(moviePoster).toHaveAttribute('src', 'example.jpg');
  });

  it('calls onClick handler when clicked', () => {
    const onClickMock = jest.fn();
    const { container } = render(<MovieTitle movie={movie} onClick={onClickMock} />);

    fireEvent.click(container.firstChild);

    expect(onClickMock).toHaveBeenCalledWith(movie);
  });
});
