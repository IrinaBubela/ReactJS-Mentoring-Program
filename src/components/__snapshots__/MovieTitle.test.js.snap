// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieTitle renders correctly 1`] = `
<div>
  <div
    class="container d-flex align-items-center justify-content-center mt-4"
  >
    <div
      class="movie-details-container col text-center"
    >
      <div
        class="poster"
      >
        <img
          alt="Example Movie"
          src="example.jpg"
        />
      </div>
      <div
        class="details mt-4"
      >
        <div
          class="name-and-genre"
        >
          <h6>
            Example Movie
          </h6>
          <p>
            Action,Adventure
          </p>
        </div>
        <div
          class="year"
        >
          <p
            class="border border-gray rounded p-1"
          >
            2021
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
